pkg/bmc/access.go

```diff
@@ -29,7 +34,19 @@ type AccessDetails interface {
 	DriverInfo(bmcCreds Credentials) map[string]interface{}
 }
 
-func getTypeHostPort(address string) (bmcType, host, port string, err error) {
+// UnknownBMCTypeError is returned when the provided BMC address cannot be
+// mapped to a driver.
+type UnknownBMCTypeError struct {
+	address string
+	bmcType string
+}
+
+func (e UnknownBMCTypeError) Error() string {
+	return fmt.Sprintf("Unknown BMC type '%s' for address %s",
+		e.bmcType, e.address)
+}
+
+func getTypeHostPort(address string) (bmcType, host, port, path string, err error) {
```

Comment:
Should we change this function name to include that it returns a path? Or should we define some type that it can return so we can keep adding fields?

Reply:
Honestly it should probably return a URL object (or even be moved to the ipmi/idrac files as separate, driver-specific New() functions) but it didn't seem worth rewriting everything that was there and works adequately.

Reply:
Yeah, I just made one function because the type logic would be the same. Using a URL object might make sense.

Reply:
Also I convinced myself that there's something to be said for constructing it from parts like this, because there's no way to sneak through weird stuff (query strings, fragment ids, username + password) that we aren't expecting.